app/domain/services/ClientRepository.scala

```diff
@@ -26,13 +26,13 @@ class DevelopmentClientRepository @Inject() (clientDao: ClientDAO)
     if (client.name.isEmpty) {
       val msg = "Client name empty"
       logger.error(msg)
```

Comment:
Error-lokitus on parempi keskittää tuonne ulkoreunalla web-apille.

Hyvä flow on tämänkaltainen:

1. Luo järkevä virheviesti
2. Heitä se exceptionin sisällä eteenpäin domainista / persistence/daosta
3. Nappaa "ulkoreunalla" eli controllerissa tai controllerien keskitetyssä exception handlerissa virheet. Tässä tapauksessa controller on hyvä, kuten tähänkin asti tehty. Katsotaan myöhemmin saadaanko keskitetty handler
4. Lokita virhe `logger.error(exception.message, exception)` <-- ensimäinen osuus printtaa meidän määritelemän virheviestin server lokille, toinen oksentaa stack tracen
4. Muunna tietyt virheet halutuiksi http status-koodin omaaviksi virheiksi ja heitä/palauta ne ulkomaailmalle.
5. Älä päästä liikaa sisäistä tietoa ulospäin serialisoidussa virhe-formaatissa